### ISSUE TITLE
`prefer-module` should not report `module` name in types

### ISSUE THREAD

**Issue by ST-DDT** [sindresorhus/eslint-plugin-unicorn#2208]
The `prefer-module` rule incorrectly marks the `module` name as forbidden in custom types/interfaces.

````txt
  18:4   error  Do not use "module"     unicorn/prefer-module
````

```js
// This should be allowed
export interface ModuleRegistry {
  [module: string]: Entry;
}
```

**Comment by fisker** [comment]
Would you like to fix it?

Add `TSIndexSignature` here

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/61e66017f0c582c8a2537ca23b2e858e1fad8908/rules/ast/is-reference-identifier.js#L4

should fix that.

**Comment by ST-DDT** [comment]
Created a PR to fix that: 

- https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2209
